interface/resources/qml/controls-uit/Tree.qml

```diff
@@ -133,11 +138,83 @@ TreeView {
             verticalCenter: parent ? parent.verticalCenter : undefined
         }
 
-        text: styleData.value
-        size: hifi.fontSizes.tableText
-        color: colorScheme == hifi.colorSchemes.light
-                   ? (styleData.selected ? hifi.colors.black : hifi.colors.baseGrayHighlight)
-                   : (styleData.selected ? hifi.colors.black : hifi.colors.lightGrayText)
+        function getComponent() {
+            if (treeView.canEdit && styleData.selected) {
+                return textFieldComponent;
+            } else {
+                return labelComponent;
+            }
+
+        }
+        sourceComponent: getComponent()
+
+        Component {
```

Comment:
Yes, because it is not part of the graphical hierarchy.
It gets dynamically switched by the loader.